models/transfers/bitcoin/transfers_bitcoin_satoshi.sql

```diff
@@ -46,21 +46,32 @@ with
         and block_time >= date_trunc('day', now() - interval '7' day)
         {% endif %}
     )
+    , transfer_btc as (
+        select any_value(type) as type, 
+            tx_id, index, 'bitcoin' as blockchain, 
+            any_value(wallet_address) as wallet_address, 
+            any_value(block_time) as block_time, 
+            any_value(block_date) as block_date, 
+            any_value(block_height) as block_height, 
+            any_value(amount_raw) as amount_raw
+        from input_transfers group by tx_id, index
+        union
+        select any_value(type) as type, 
+            tx_id, index, 'bitcoin' as blockchain, 
+            any_value(wallet_address) as wallet_address, 
+            any_value(block_time) as block_time, 
+            any_value(block_date) as block_date, 
+            any_value(block_height) as block_height, 
+            any_value(amount_raw) as amount_raw
+        from output_transfers group by tx_id, index
+    )
 
-select any_value(type) as type, 
-    tx_id, index, 'bitcoin' as blockchain, 
-    any_value(wallet_address) as wallet_address, 
-    any_value(block_time) as block_time, 
-    any_value(block_date) as block_date, 
-    any_value(block_height) as block_height, 
-    any_value(amount_raw) as amount_raw
-from input_transfers group by tx_id, index
-union
-select any_value(type) as type, 
-    tx_id, index, 'bitcoin' as blockchain, 
-    any_value(wallet_address) as wallet_address, 
-    any_value(block_time) as block_time, 
-    any_value(block_date) as block_date, 
-    any_value(block_height) as block_height, 
-    any_value(amount_raw) as amount_raw
-from output_transfers group by tx_id, index
\ No newline at end of file
+SELECT t.type, t.tx_id, t.index, t.blockchain,
+    t.wallet_address, t.block_time, t.block_date,
+    t.block_height, t.amount_raw,
+    -1 * t.amount_raw * p.price as amount_transfer_usd
+FROM transfer_btc t
+LEFT JOIN {{ source('prices', 'usd') }} p
```

Comment:
please add an incremental block in this join condition, so that prices only reads the last 7 days, similar to above on your source objects

Reply:
I'm done, please review again

Reply:
if we look at the logs of the CI test run [here](https://github.com/duneanalytics/spellbook/actions/runs/6123852890/job/16622650058?pr=4276#step:17:18), it's odd that the incremental step (`dbt run incremental model(s) if applicable`) is taking longer than the historical run step (`dbt run initial model(s)`)

that should rarely, if at all, happen due to data volumes being read from the source. we will need to look closer at why the latest changes have made this happen

cc @longcnttbkhn

Reply:
random thoughts:
- could the union in line 58 be a union all?
- could join to prices be an inner join?
- missing filters elsewhere?

Reply:
random thoughts:
- could the union in line 58 be a union all?
- could join to prices be an inner join?
- missing filters elsewhere?

Reply:
I will check this issue

Reply:
@jeff-dude  I converted the `transfers_bitcoin_satoshi` table into a view (similar to `transfer_ethereum_erc20`), historical data processing time decreased from 5 minutes to 2 minutes, incremental processing time from 11 minutes to 3 minutes. Do you think this is okay?

Reply:
making it a view will definitely make it faster to build, as it won't actually run the query, just tell the view what logic to use when ran against on the app. this will make your queries on the app slower, most likely. it's a tradeoff.

let me take a quick look at the old incremental logic and see if anything stands out. if not, it may just be how this data is stored in our data lake and there isn't anything we can do to speed up the processing.

i think at the end of the day, your life will be easier if this goes back to incremental, and we will have to eat the processing time on our side. it also could improve in prod post-merge, not sure yet.

i'll try and push a commit or two here in a few 👍

Reply:
i'm not sure there is anything else we can do. it must be that dunesql isn't as performant on the btc address data types, since they are not varbinary like the EVM chains (varchar on btc data)

i think we can mark this as ready for merge as-is

cc @aalan3 for awareness on btc spells here and the performance related on incremental runs